book/src/scalable_lists.md

```diff
@@ -0,0 +1,179 @@
+As a first step it allows us to remove the "bind" step.
+Now let us see how the "setup" step now works.
+
+<span class="filename">Filename: listings/scalable_lists/3/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/3/main.rs:factory_setup}}
+```
+
+An expression describe references to values.
+So when we create a [`ConstantExpression`](../git/docs/gtk4/struct.ConstantExpression.html) of `list_item`, we create a reference to a `ListItem`.
+We then create a [`PropertyExpression`](../git/docs/gtk4/struct.PropertyExpression.html) to get a reference to the "item" property of `list_item`.
+With another `PropertyExpression` we get a reference to the "number" property of the "item" property of `list_item`.
+That already makes the first power of expressions obvious: it allows nested relationships.
+Finally, we bind "number" to "label".
+In pseudo code that would be `list_item->item->number == label->label`.
+
+It is worth noting that at the "setup" stage there is no way of knowing which list item belongs to which label.
+Simply because this changes as we scroll through the list.
+And this is the power of expressions.
+We do not have to define a fixed relationship, the object and properties might not even exist yet.
+We just had to tell it to change the label whenever the number changes that belongs to it.
+That we way, we also do not face the problem that multiple labels are bound to the same number.
+
+Whenever we now activate a label, the number gets visibly changed.
+However, that is still not everything we can do.
+We can, for example, filter our model to only allow even numbers.
+
+<span class="filename">Filename: listings/scalable_lists/4/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/4/main.rs:filter}}
+```
+
+Additionally, we can reverse the sorting of our model.
+
+<span class="filename">Filename: listings/scalable_lists/4/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/4/main.rs:sorter}}
+```
+
+In order to ensure that our filter and sorter gets updated when we modify the numbers, we call the `changed` method on them.
+
+<span class="filename">Filename: listings/scalable_lists/4/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/scalable_lists/4/main.rs:activate}}
+```
+
+After our changes, the application looks like this.
+
+<div style="text-align:center"><img src="img/scalable_lists_demo_2.png"/></div>
+
+
+We now know how to handle huge amount of data with the help of models such as [`gio::ListStore`](http://gtk-rs.org/docs/gio/struct.ListStore.html) and widgets like [`ListView`](../git/docs/gtk4/struct.ListView.html), [`ColumnView`](../git/docs/gtk4/struct.ColumnView.html) or [`GridView`](../git/docs/gtk4/struct.GridView.html).
+Please note that you do not have to have thousands of element until it makes sense to use this pattern.
```

Comment:
```suggestion
Please note that you do not have to have thousands of element until it makes sense to use this pattern.
```
are you sure about this?

Reply:
Why not use it when you only have ~100?

Reply:
why use them when a listbox can do the job and is much simpler to setup?

Reply:
the models are usable in listbox/flowbox as well btw. It's just you lose a ton of flexibility that was added to the *View widgets. but yes, starting the chapter with Listbox/FlowBox and having a section on scalable ones where you introduce the use of the View widgets would be much better.